code/modules/reagents/chemistry/holder.dm

```diff
@@ -499,32 +497,27 @@ var/const/INGEST = 2
 			can_process = 1
 	return can_process
 
-/datum/reagents/proc/reaction(atom/A, method=TOUCH, volume_modifier = 1)
-	switch(method)
-		if(TOUCH)
-			for(var/datum/reagent/R in reagent_list)
-				if(isliving(A))
-					var/check = reaction_check(A, R)
-					if(!check)
-						continue
-					else
-						R.reaction_mob(A, TOUCH, R.volume*volume_modifier)
-				if(isturf(A))
-					R.reaction_turf(A, R.volume*volume_modifier)
-				if(isobj(A))
-					R.reaction_obj(A, R.volume*volume_modifier)
-		if(INGEST)
-			for(var/datum/reagent/R in reagent_list)
-				if(isliving(A))
-					var/check = reaction_check(A, R)
-					if(!check)
-						continue
-					else
-						R.reaction_mob(A, INGEST, R.volume*volume_modifier)
-				if(isturf(A))
-					R.reaction_turf(A, R.volume*volume_modifier)
-				if(isobj(A))
-					R.reaction_obj(A, R.volume*volume_modifier)
+/datum/reagents/proc/reaction(atom/A, method = TOUCH, volume_modifier = 1)
+	var/react_type
+	if(isliving(A))
+		react_type = "LIVING"
+	else if(isturf(A))
+		react_type = "TURF"
+	else if(istype(A, /obj))
```

Comment:
@Fox-McCloud That's not *quite* accurate- It will return true for any actual /obj and any *custom* branches of /atom/movable- which is only lighting overlays.